### DISCUSSION TITLE
bypassing cloudflare captcha gives error

### DISCUSSION THREAD

**Discussion by riyazuddin-1** [seleniumbase/SeleniumBase#2822] [Q&A] [answered]
I have been trying to bypass the cloudflare captcha, and it works, but it throws an exception:
```
seleniumbase.common.exceptions.NoSuchElementException: Message:
 Element {img#captcha-success} was not present after 3 seconds!
```

`from seleniumbase import SB

def open_the_turnstile_page(sb):
    url = "remotive.com"
    sb.driver.uc_open_with_reconnect(url, reconnect_time=2)

def click_turnstile_and_verify(sb):
    sb.driver.switch_to_frame("iframe")
    sb.driver.uc_click("span")
    sb.assert_element("img#captcha-success", timeout=3)

with SB(uc=True, test=True) as sb:
    open_the_turnstile_page(sb)
    try:
        click_turnstile_and_verify(sb)
    except Exception:
        open_the_turnstile_page(sb)
        click_turnstile_and_verify(sb)
    sb.set_messenger_theme(location="top_left")
    sb.post_message("SeleniumBase wasn't detected", duration=3)`
    
    If I remove the 3rd line in the function `click_turnstile_and_verify()` scraping is not working
    Can anybody help?

**Comment by mdmintz** [comment, marked as answer]
Try the example script: [SeleniumBase/examples/raw_turnstile.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_turnstile.py)

On the website you gave, I didn't encounter any CAPTCHAs. You also said that "it works", but threw an exception when looking for the success message. The success message can be different on different websites. Adjust your scripts accordingly.

**Reply by riyazuddin-1** [reply]
This is the same code from the example script, I just changed the website link. The only problem I am facing here is with the success message.
I am using beautifulSoup after this for parsing, but couldn't proceed further because of the following error
`seleniumbase.common.exceptions.NoSuchElementException: Message:
 Element {img#captcha-success} was not present after 3 seconds!`
 
![Screenshot 2024-06-03 194045](https://github.com/seleniumbase/SeleniumBase/assets/116349117/c9ff186b-a026-4d4a-a25b-8bdff3cdcd20)
![Screenshot 2024-06-03 194236](https://github.com/seleniumbase/SeleniumBase/assets/116349117/765c7fd5-d063-4fa1-8600-c24078e66846)
![Screenshot 2024-06-03 193956](https://github.com/seleniumbase/SeleniumBase/assets/116349117/f4863ee0-55ff-4b3d-8fc9-a317d1dc6642)

**Reply by mdmintz** [reply]
That's a different type of Cloudflare CAPTCHA. The one that you don't need to click to bypass. Therefore, the success message is different, and won't last long enough for you to verify. Verify success by seeing that you made it to the page that you wanted to reach, such as in this example: [SeleniumBase/examples/raw_uc_mode.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_uc_mode.py).
